src/profile-logic/marker-data.js

```diff
@@ -119,71 +119,184 @@ export function getSearchFilteredMarkerIndexes(
   getMarker: (MarkerIndex) => Marker,
   markerIndexes: MarkerIndex[],
   markerSchemaByName: MarkerSchemaByName,
-  searchRegExp: RegExp | null,
+  searchRegExps: MarkerRegExps | null,
   categoryList: CategoryList
 ): MarkerIndex[] {
-  if (!searchRegExp) {
+  if (!searchRegExps) {
     return markerIndexes;
   }
+
+  const hasPositiveRegexps =
+    searchRegExps.generic ||
+    Array.from(searchRegExps.fieldMap.entries()).some(
+      ([_, val]) => val.positive !== null
+    );
+  const hasNegativeRegexps = Array.from(searchRegExps.fieldMap.entries()).some(
+    ([_, val]) => val.negative !== null
+  );
+
+  const newMarkers: MarkerIndex[] = [];
+  for (const markerIndex of markerIndexes) {
+    const marker = getMarker(markerIndex);
+    let includeMarker = true;
+    if (hasPositiveRegexps) {
+      includeMarker = positiveFilterMarker(
+        marker,
+        markerSchemaByName,
+        searchRegExps,
+        categoryList
+      );
+    }
+
+    if (includeMarker && hasNegativeRegexps) {
+      includeMarker = negativeFilterMarker(
+        marker,
+        markerSchemaByName,
+        searchRegExps,
+        categoryList
+      );
+    }
```

Comment:
it could be good to document the possible behaviors in some comment, depending on all combinations of hasPositiveRegexps and hasNegativeRegexps